### PR TITLE
ed25519 ssh key support for compute and storage account local user

### PR DESCRIPTION
**Pull request by gerrytan**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

Adds support for ed25519 ssh key for compute resource ([already supported in az cli](https://learn.microsoft.com/en-us/cli/azure/sshkey?view=azure-cli-latest#az-sshkey-create-optional-parameters)) and storage account local user resource.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

- Manual testing: can deploy a linux vm with ed25519 ssh key, and connect to it via bastion - [main.tf](https://gist.github.com/gerrytan/1b2f72bf93c12a84650df5ae1658b312#file-main-tf) - [terraform apply log](https://microsoftapc-my.sharepoint.com/:u:/g/personal/tangerry_microsoft_com/EbHBS8TiGjdPgdeykZv0FAEB3G76iTMvNP8oTv6ocLbclg?e=mLA4TQ) ✅
- `make acctests SERVICE='compute' TESTARGS='-run=TestAccLinuxVirtualMachine_auth' TESTTIMEOUT='120m'` - [output log](https://microsoftapc-my.sharepoint.com/:u:/g/personal/tangerry_microsoft_com/EQjJGpberEZBh_XaWGUsnTABjWqkSRbA_EVxBpp8pvFZxA?e=4aIRDK) ✅
- `make acctests SERVICE='compute' TESTARGS='-run=TestAccOrchestratedVirtualMachineScaleSet_' TESTTIMEOUT='120m'` - TBA
- `make acctests SERVICE='storage' TESTARGS='-run=TestAccLocalUser_' TESTTIMEOUT='120m'` - [output log](https://microsoftapc-my.sharepoint.com/:u:/g/personal/tangerry_microsoft_com/EcmJxO7ZAshPjbkti1imf_4BppMd5QHUgyDYQQ3NcbvDZQ?e=3ii0iV) ✅

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_linux_virtual_machine`, `azurerm_storage_account_local_user` - adds support for ed25519 ssh key [GH-26587]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)

#26587 

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/storage/storage_account_local_user_resource.go [hunk 1]
```diff
@@ -93,7 +92,7 @@ func (r LocalUserResource) Arguments() map[string]*pluginsdk.Schema {
 					"key": {
 						Type:             pluginsdk.TypeString,
 						Required:         true,
-						ValidateFunc:     computevalidate.SSHKey,
+						ValidateFunc:     validate.SSHKey,
```
Comment (magodo) [inline internal/services/storage/storage_account_local_user_resource.go:95]: Have you verified that the storage account local user still doesn't support ED25519 public key yet?

**Review by magodo — COMMENTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline, but this is mostly looking good to me 👍

File: internal/services/storage/storage_account_local_user_resource.go (hunk 1 above)
Comment (gerrytan) [reply internal/services/storage/storage_account_local_user_resource.go]: Actually it supports ed25519. I have reverted this diff so now the PR covers both compute and storage account local user. Commit ref: dfa49be2f5

**Review by catriona-m — APPROVED** [review]
Thanks @gerrytan LGTM!

**Pull request merged**